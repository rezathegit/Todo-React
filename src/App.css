* {
  margin:0;
  padding:0;
  box-sizing: border-box;
}
body {
  background-image: linear-gradient(120deg, #fdfbfb 0%, #ebedee 100%);
  color: darkgrey;
  font-family: 'Roboto', sans-serif;
  min-height: 100vh;
}

header {
  font-size: 1.5rem;
}
header, form {
  height: 7rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

.todo-input,
.todo-button {
  border: none;
  background: white;
  padding: 0.5rem;
  font-size: 2rem;
}

.todo-button {
  color: darkgrey;
  background: white;
  cursor: pointer;
  transition: all 0.3s ease;
}

.todo-button:hover {
  background: orange;
  color: white;
}

.todo-container {
  justify-content: center;
  align-items: center;
}
.todo-list {
  min-width: 30%;
  list-style: none;
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
  height: 450px;
}

.todo {
  margin: 1rem;
  background: white;
  font-size: 1.5rem;
  color: black;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: all 1s ease;
}
.todo li {
  flex: 1;
}

.done-btn {
background: white;
color: darkgrey;
border: none;
padding: 1rem;
cursor: pointer;
font-size: 1rem;

}
.done-btn:hover {
  background: green;
  color: white;
}

.delete-btn {
background: white;
color: darkgrey;
border: none;
padding: 1rem;
cursor: pointer;
font-size: 1rem;
  
  }
.delete-btn:hover {
  background: red;
  color: white;
  }

.completed {
  text-decoration: line-through;
  opacity: 0.3;
  color: palevioletred;
  }
.todo-item {
  padding: 0rem 0.4rem;
  overflow: hidden;
  overflow-wrap: break-word;
  }

.input-error {
  color: red;
  text-align: center;
  font-weight: 300;
  font-size: 1rem;
}