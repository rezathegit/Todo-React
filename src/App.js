import React, { useState } from 'react';
import './App.css';
import TodoForm from './components/TodoForm';
import TodoList from './components/TodoList';

function App() {
  const [todoInput, setTodoInput] = useState('');
  const [todos, setTodos] = useState([]);
  const [inputError, setInputError] = useState('');

  return (
    <div className="App">
      <header>
        <h1>Todo App</h1>
        <span className="material-icons-outlined">face</span>
      </header>
      <TodoForm
        todos={todos}
        setTodos={setTodos}
        todoInput={todoInput}
        setTodoInput={setTodoInput}
        setInputError={setInputError} />
      <p className="input-error">{inputError}</p>
      <TodoList todos={todos} setTodos={setTodos} />
    </div>
  );
}

export default App;
