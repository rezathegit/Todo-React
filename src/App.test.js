import { render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('App test suite', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('tests that App is rendered', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toBeTruthy();
  });

  it("tests if div element has the class 'App'", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toHaveClass('App');
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
