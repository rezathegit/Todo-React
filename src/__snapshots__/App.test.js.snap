// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      Todo App
    </h1>
    <span
      className="material-icons-outlined"
    >
      face
    </span>
  </header>
  <form>
    <input
      className="todo-input"
      onChange={[Function]}
      placeholder="What to do?"
      type="text"
      value=""
    />
    <button
      className="todo-button"
      onClick={[Function]}
      type="submit"
    >
      <span
        className="material-icons-outlined"
      >
        add
      </span>
    </button>
  </form>
  <p
    className="input-error"
  >
    
  </p>
  <div
    className="todo-container"
  >
    <ul
      className="todo-list"
    />
  </div>
</div>
`;
